### PR TITLE
Fixed mb way component

### PR DESCRIPTION
**Pull request by mohammedDehairy**
## Summary
 1. Remove MBWay email field.
 2. Fixe an issue with DropInNavigationController, where the preferred content size of content view controllers can be wrong.
 3. Made the Phone number field in MBWayComponent accept any phone number from anywhere in the world.
 4. Fixed Unit tests to reflect that any phone number is accepted.
 5. Created a generic RegularExpressionValidator and made EmailValidator and PhoneNumberValidator inherit from it.

## Tested scenarios
Covered the new logic with unit tests and modified the old unit tests.

### CODE REVIEW

File: Adyen/Validators/RegularExpressionValidator.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2020 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// Validates a phone number.
```
Comment (descorp) [inline Adyen/Validators/RegularExpressionValidator.swift:9]: Comment is outdated :D

File: AdyenDropIn/Presentation/DropInNavigationController.swift [hunk 2]
```diff
@@ -49,13 +49,15 @@ internal final class DropInNavigationController: UINavigationController {
     // MARK: - Keyboard
     
     @objc private func keyboardWillChangeFrame(_ notification: NSNotification) {
-        guard let topViewController = topViewController as? WrapperViewController,
-            topViewController.requiresInput,
-            let bounds = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect
-        else { return }
+        if let bounds = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect {
```
Comment (descorp) [inline AdyenDropIn/Presentation/DropInNavigationController.swift:52]: I believe there was cases when updating `keyboardRect ` is not needed like stored card payment.

File: Adyen/Formatters/PhoneNumberFormatter.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//
+// Copyright (c) 2020 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+open class PhoneNumberFormatter: Adyen.Formatter {
```
Comment (descorp) [inline Adyen/Formatters/PhoneNumberFormatter.swift:10]: Are we going to use this in Qiwi payment as well?

File: AdyenDropIn/Presentation/DropInNavigationController.swift (hunk 2 above)
Comment (mohammedDehairy) [reply AdyenDropIn/Presentation/DropInNavigationController.swift:52]: stored payment methods works fine, also i had a problem with AwaitComponent where the preferred height takes into account the keyboard height even though the await view has no fields, and when investigated, found out the thekeyboardRect wasn't reset when leaving the MBWay form, hence the change.

File: Adyen/Formatters/PhoneNumberFormatter.swift (hunk 3 above)
Comment (mohammedDehairy) [reply Adyen/Formatters/PhoneNumberFormatter.swift:10]: don't think so, no need to change Qiwi

File: Adyen/Validators/RegularExpressionValidator.swift (hunk 1 above)
Comment (mohammedDehairy) [reply Adyen/Validators/RegularExpressionValidator.swift]: 👍  fixed

**Review by descorp — APPROVED** [review]

**Pull request merged**